lib/neo/rails/collection_presenter.rb

```diff
@@ -0,0 +1,53 @@
+module Neo
+  module Rails
+    module CollectionPresenter
+      def self.included(base)
+        base.extend(ClassMethods)
+
+        base.class_eval do
+          include Enumerable
+
+          delegate :last, :size, :each, :to => :@presenters
+        end
+      end
+
+      def initialize(presenters)
+        @presenters = presenters
+      end
+
+      def get_by_index(index)
+        @presenters[index]
+      end
+
+      def view_context
+        Neo::Rails::Presenter.view_context
+      end
+
+      module ClassMethods
+        def build(objects)
```

Comment:
What do you think about allowing explicitly passing the `presenter_class`:

```ruby
class UserPresenter
  def initialize(object); end
end

class Admin::UserCollectionPresenter
  include Neo::Rails::CollectionPresenter 
end

collection_presenter = Admin::UserCollectionPresenter.build([], UserPresenter)
```

Or the possibility to use block as factory? This would also allow to use different item presenters per object:

```ruby
collection_presenter = Admin::UserCollectionPresenter.build([]) do |object|
  object.nil? ? NullUserPresenter.new : UserPresenter.new(object)
end
```




I don't want to over engineer this, but I known already at least one point to use a custom presenters in a collection.